admin/commands/pmm/server/docker/upgrade.go

```diff
@@ -71,9 +73,20 @@ func (u *upgradeResult) String() string {
 	return "ok"
 }
 
+func New(l *logrus.Entry, waitBeforeContainerStop time.Duration) *UpgradeCommand {
+	return &UpgradeCommand{
+		l:                       l,
+		waitBeforeContainerStop: waitBeforeContainerStop,
+	}
+}
+
 // RunCmdWithContext runs upgrade command.
-func (c *UpgradeCommand) RunCmdWithContext(ctx context.Context, globals *flags.GlobalFlags) (commands.Result, error) { //nolint:unparam
-	logrus.Info("Starting PMM Server upgrade via Docker")
+func (c *UpgradeCommand) RunCmdWithContext(ctx context.Context, _ *flags.GlobalFlags) (commands.Result, error) { //nolint:unparam
+	if c.l == nil {
+		c.l = logrus.NewEntry(logrus.StandardLogger())
+	}
```

Comment:
Is it really needed?

Reply:
The `UpgradeCommand` is used both by `pmm` and `pmm-server-upgrade`.
The `pmm-server-upgrade` redirects logs to a file and provides its own instance of logger.
`pmm` on the other hand does not initialize the logger when the command runs, thus the initialization.

Reply:
`New` is not used when Kong calls this command. We need this for Kong to properly init logger.